### PR TITLE
Enable/disable rule detail flyout

### PR DESCRIPTION
**Pull request by swetha571**
# Type of change? 

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Enable/Disable rule details flyout

...

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/components/ruleOverview/ruleOverview.js [hunk 1]
```diff
@@ -64,10 +64,8 @@ class RuleOverviewFlyout extends Component {
                 <div className="clearfix"/>
                 <div className="rule-overview-header">
                     <div className="toggle-button">
-                        <div onClick={this.onToggleStatus}
-                             className={`icon ${!this.state.enabled ? 'icon-enable' : 'icon-disable'}`}/>
-                        <div
-                            className="text">{this.state.enabled ? lang.ENABLESELECTEDRULE : lang.DISABLESELECTEDRULE}</div>
+                        <div className={`icon ${this.state.enabled ? 'icon-disable': 'icon-enable'}`}/>
+                        <div className="text">{this.state.enabled ?  lang.DISABLESELECTEDRULE : lang.ENABLESELECTEDRULE}</div>
```
Comment (spryor) [inline src/components/ruleOverview/ruleOverview.js]: There is an extra space after the `?`

File: src/components/ruleOverview/ruleOverview.js [hunk 2]
```diff
@@ -17,15 +17,15 @@ class RuleOverviewFlyout extends Component {
         };
```
Comment (dluc) [inline src/components/ruleOverview/ruleOverview.js]: is this equivalent to:
```
this.state = {
  enabled: props.rules.some(rule => !rule.Enabled)
}
```

File: src/components/ruleOverview/ruleOverview.js [hunk 3]
```diff
@@ -17,15 +17,15 @@ class RuleOverviewFlyout extends Component {
...
-        this.setState({enabled: !this.state.enabled});
-    };
-
     onSave = () => {
         let promises = [];
         let updatedRules = [];
         this.props.rules.forEach(rule => {
-            if (rule.Enabled !== this.state.enabled) {
+            if (rule.Enabled === this.state.enabled) {
+                rule.Enabled = !this.state.enabled;
```
Comment (dluc) [inline src/components/ruleOverview/ruleOverview.js]: is this equivalent to `rule.Enabled = false` ? if so I'd change it

File: src/components/ruleOverview/ruleOverview.js (hunk 2 above)
Comment (swetha571) [reply src/components/ruleOverview/ruleOverview.js]: No. If the rule is already disabled - then when we open the flyout, we see the enable toggle.
If the rule is already ENABLED -  then when we open the flyout , we see the disable toggle with out user toggles them.

File: src/components/ruleOverview/ruleOverview.js [hunk 4]
```diff
@@ -17,15 +17,15 @@ class RuleOverviewFlyout extends Component {
...
+                updatedRules.push(rule);
```
Comment (spryor) [inline src/components/ruleOverview/ruleOverview.js]: The rule is pushed into the array and the promise is pushed into its array regardless of the if condition. Move these values out of the if and do them all the time.

**Pull request merged**